Userland/Services/LookupServer/DNSAnswer.cpp

```diff
@@ -18,15 +18,16 @@ DNSAnswer::DNSAnswer(const DNSName& name, DNSRecordType type, DNSRecordClass cla
     , m_record_data(record_data)
     , m_mdns_cache_flush(mdns_cache_flush)
 {
-    auto now = time(nullptr);
-    m_expiration_time = now + m_ttl;
-    if (m_expiration_time < now)
-        m_expiration_time = 0;
+    time(&m_received_time);
 }
 
 bool DNSAnswer::has_expired() const
 {
-    return time(nullptr) >= m_expiration_time;
+    auto expiration_time = m_received_time + m_ttl;
+    auto now = time(nullptr);
+    if (expiration_time < now)
+        expiration_time = 0;
+    return now >= expiration_time;
```

Comment:
What does the
```cpp
if (expiration_time < now)
     expiration_time = 0;
```
part do?

Reply:
Now that I think about - nothing, really. I moved that from the constructor. If `expiration_time` overflows it'll be less than `now` anyway. I'll remove that check.

Reply:
Well, that seems unnecessary here seeing how `m_received_time` is a 64-bit integer (at around somewhere 2 ** 32 at the moment) and `m_ttl` is an unsigned 32-bit integer.

Reply:
Yeah; not saying you should do it here; just that this is how overflow checking is to be done if you're doing it, not with the `a + b < a` thing.